thumbnail-service/src/main/java/no/sikt/nva/thumbnail/thumbnailer/FFMpegThumbnailer.java

```diff
@@ -0,0 +1,71 @@
+package no.sikt.nva.thumbnail.thumbnailer;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import net.bramp.ffmpeg.FFmpeg;
+import net.bramp.ffmpeg.FFmpegExecutor;
+import net.bramp.ffmpeg.FFprobe;
+import net.bramp.ffmpeg.builder.FFmpegBuilder;
+import no.sikt.nva.thumbnail.AbstractThumbnailer;
+import no.sikt.nva.thumbnail.util.ImageResizer;
+
+public class FFMpegThumbnailer extends AbstractThumbnailer {
+
+    private final FFmpeg ffmpeg;
+    private final FFprobe ffprobe;
+
+    public static final String TMP_VIDEOSNAP_PNG = "/tmp/videosnap.png";
+
+    public FFMpegThumbnailer(FFmpeg ffmpeg, FFprobe ffprobe) {
+        super();
+        this.ffmpeg = ffmpeg;
+        this.ffprobe = ffprobe;
+    }
+
+    @Override
+    public void generateThumbnail(File input, File output) throws IOException {
+        //
+        File partiallyProcessed = new File(TMP_VIDEOSNAP_PNG);
+        FFmpegBuilder builder = new FFmpegBuilder()
```

Comment:
use variables and constants instead of comments.

Suggestion:
```
ublic class FFMpegThumbnailer extends AbstractThumbnailer {
    
    public static final boolean OVERRIDE_OUTPUT = true;
    private final FFmpeg ffmpeg;
    private final FFprobe ffprobe;
    
    public FFMpegThumbnailer(FFmpeg ffmpeg, FFprobe ffprobe) {
        super();
        this.ffmpeg = ffmpeg;
        this.ffprobe = ffprobe;
    }
    
    public static final File TEMPORARY_FILE_FOR_PARTIALLY_GENERATED_RESULTS =
        new File(String.format("/tmp/%s.png", UUID.randomUUID()));
    
    @Override
    public void generateThumbnail(File input, File output) throws IOException {
        encodeInput(input);
        resizeToThumbnailSpecs(output);
        TEMPORARY_FILE_FOR_PARTIALLY_GENERATED_RESULTS.deleteOnExit();
    }
    
    private void encodeInput(File input) {
        var config = setupEncodingConfiguration(input);
        runOnePassEncode(config);
    }
    
    private static FFmpegBuilder setupEncodingConfiguration(File input) {
        var fileOrFFmpegProbeResult = input.getAbsolutePath();
        var destinationFilename = TEMPORARY_FILE_FOR_PARTIALLY_GENERATED_RESULTS.getAbsolutePath();
        return new FFmpegBuilder()
                   .setInput(fileOrFFmpegProbeResult)
                   .overrideOutputFiles(OVERRIDE_OUTPUT)
                   .addOutput(destinationFilename)
                   .setStartOffset(3, TimeUnit.SECONDS)
                   .setFrames(1)
                   .done();
    }
    
    private void runOnePassEncode(FFmpegBuilder builder) {
        var executor = new FFmpegExecutor(ffmpeg, ffprobe);
        executor.createJob(builder).run();
    }
    
    //ffmpeg doesn't operate with mimetypes but with formats.
    //Here are some of the formats converted to mimetypes:
    @Override
    public List<String> getAcceptedMimeTypes() {
        return List.of("application/vnd.ms-asf",
            "application/ffmpeg",
            "video/x-msvideo",
            "video/x-flv",
            "video/webm",
            "video/mpeg",
            "video/x-m4v",
            "video/mp4",
            "video/ogg",
            "video/x-matroska",
            "video/mpeg",
            "video/quicktime"
        );
    }
    
    private void resizeToThumbnailSpecs(File output) throws IOException {
        ImageResizer imageResizer = new ImageResizer(thumbWidth, thumbHeight,
            TEMPORARY_FILE_FOR_PARTIALLY_GENERATED_RESULTS);
        imageResizer.writeThumbnailToFile(output);
    }
}
```

Reply:
Had to change this a bit, tests would not run with dynamic generated partially processed filename without injecting the filename. Added the partiallyProcessedFilename to the thumbnail initializer.